src/libsync/discovery.cpp

```diff
@@ -896,46 +896,68 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
-            }
+        if (!localEntry.isVirtualFile && !localEntry.isDirectory) {
+            return;
+        }
 
-            // must be a file placeholder or an online-only folder placeholder
-            if (isFilePlaceHolder || isFolderPlaceholder) {
-                if (isFolderPlaceholder) {
-                    qCInfo(lcDisco) << "Wiping virtual folder without db entry for" << path._local;
-                } else {
-                    qCInfo(lcDisco) << "Wiping virtual file without db entry for" << path._local;
-                }
-                item->_instruction = CSYNC_INSTRUCTION_REMOVE;
-                item->_direction = SyncFileItem::Down;
-                // this flag needs to be unset, otherwise a folder would get marked as new in the processSubJobs
-                _childModified = false;
-                if (isFolderPlaceholder && _discoveryData) {
-                    emit _discoveryData->addErrorToGui(SyncFileItem::SoftError, tr("Conflict when uploading a folder. It's going to get cleared!"), path._local);
-                }
-            } else {
-                if (localEntry.isDirectory && !isFolderPlaceholder) {
-                    qCInfo(lcDisco) << "Virtual directory without db entry for" << path._local << "but it contains hydrated file(s), so let's keep it and reupload.";
-                    if (_discoveryData) {
-                        emit _discoveryData->addErrorToGui(SyncFileItem::SoftError, tr("Conflict when uploading some files to a folder. Those, conflicted, are going to get cleared!"), path._local);
-                    }                    
-                    return;
-                }
-                qCWarning(lcDisco) << "Virtual file without db entry for" << path._local
-                                   << "but looks odd, keeping";
-                item->_instruction = CSYNC_INSTRUCTION_IGNORE;
+        Q_ASSERT(item->_instruction == CSYNC_INSTRUCTION_NEW);
+        if (item->_instruction != CSYNC_INSTRUCTION_NEW) {
+            qCWarning(lcDisco) << "Trying to wipe a virtual item" << path._local << " with item->_instruction" << item->_instruction;
+            return;
+        }
+
+        // must be a dehydrated placeholder
+        const bool isFilePlaceHolder = !localEntry.isDirectory && _discoveryData->_syncOptions._vfs->isDehydratedPlaceholder(_discoveryData->_localDir + path._local);
+
+        // either correct availability, or a result with error if the folder is new or otherwise has no availability set yet
+        const auto folderAvailability = localEntry.isDirectory ? _discoveryData->_syncOptions._vfs->availability(path._local) : Vfs::AvailabilityError::NoSuchItem;
+
+        const auto folderPinState = localEntry.isDirectory ? _discoveryData->_syncOptions._vfs->pinState(path._local) : PinState::Unspecified;
+
+        if (!isFilePlaceHolder && !folderAvailability && !folderPinState) {
```

Comment:
see my previous comment, I would have an hard time explaining what this check is doing
please make it more obvious such that a comment would not be needed